packages/foreman/nodejs-history/nodejs-history.spec

```diff
@@ -16,38 +19,49 @@ Source5: https://registry.npmjs.org/value-equal/-/value-equal-0.4.0.tgz
 Source6: https://registry.npmjs.org/resolve-pathname/-/resolve-pathname-2.2.0.tgz
 Source7: https://registry.npmjs.org/regenerator-runtime/-/regenerator-runtime-0.13.2.tgz
 Source8: https://registry.npmjs.org/js-tokens/-/js-tokens-4.0.0.tgz
-Source9: %{name}-%{version}-registry.npmjs.org.tgz
-BuildRequires: nodejs-packaging
+Source9: nodejs-%{npm_name}-%{version}-registry.npmjs.org.tgz
+
+Requires:       rh-nodejs10-nodejs
+BuildRequires:  rh-nodejs10-nodejs-devel
+BuildRequires:  rh-nodejs10-build
+BuildRequires:  rh-nodejs10-npm
```

Comment:
We can choose to not hard-code these if we want to make the `tfm` dependent on `rh-nodejs10` and define macros therein like we do for Rails.

Reply:
If we make tfm dependent on nodejs, would that require nodejs on every deployment rather than the build? I'd still prefer to use macros and keep it in the build env, but do the dependency in mock and koji. Possibly via a tfm-nodejs subpackage which in turn depends on the correct nodejs. I think this will make it easier when we also want to support EL8 modules.

Reply:
We would omit nodejs from the tfm-runtime package and relegate it only to build level packages. I think this would achieve what you want without having to spawn out a separate package but still provide all the same macros in all the environments we typically expect them in.

Reply:
I'll open a PR to show my thinking.